pkg/fqdn/service/service_test.go

```diff
@@ -0,0 +1,260 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package service
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net/netip"
+	"testing"
+
+	"github.com/cilium/hive/cell"
+	"github.com/cilium/hive/hivetest"
+	"github.com/stretchr/testify/require"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials/insecure"
+
+	"github.com/cilium/cilium/pkg/clustermesh/types"
+	"github.com/cilium/cilium/pkg/defaults"
+	"github.com/cilium/cilium/pkg/endpoint"
+	"github.com/cilium/cilium/pkg/endpointmanager"
+	"github.com/cilium/cilium/pkg/fqdn/messagehandler"
+	"github.com/cilium/cilium/pkg/fqdn/namemanager"
+	"github.com/cilium/cilium/pkg/hive"
+	"github.com/cilium/cilium/pkg/ipcache"
+	"github.com/cilium/cilium/pkg/option"
+	"github.com/cilium/cilium/pkg/policy"
+	"github.com/cilium/cilium/pkg/policy/api"
+	"github.com/cilium/cilium/pkg/testutils"
+	testidentity "github.com/cilium/cilium/pkg/testutils/identity"
+	"github.com/cilium/cilium/pkg/time"
+
+	pb "github.com/cilium/cilium/api/v1/standalone-dns-proxy"
+)
+
+func TestFQDNDataServer(t *testing.T) {
+
+	test := map[string]struct {
+		port                     int
+		enableL7Proxy            bool
+		enableStandaloneDNSProxy bool
+		serverPort               int
+		err                      error
+	}{
+		"Successfully running the server": {
+			port:                     1234,
+			enableL7Proxy:            true,
+			enableStandaloneDNSProxy: true,
+			serverPort:               40045,
+			err:                      nil,
+		},
+		"Failure on running the server due to invalid server port": {
+			port:                     1234,
+			serverPort:               0,
+			enableStandaloneDNSProxy: true,
+			enableL7Proxy:            true,
+			err:                      errors.New("listen tcp: address -1: invalid port"),
+		},
+	}
+
+	for scenario, tt := range test {
+		t.Run(scenario, func(t *testing.T) {
+
+			h := hive.New(
+				cell.Module(
+					"test-fqdn-grpc-server",
+					"Test FQDN gRPC server",
+					cell.Config(defaultConfig),
+					cell.Provide(
+						func() endpointmanager.EndpointManager {
+							return endpointmanager.New(&dummyEpSyncher{}, nil, nil, nil, nil)
+						},
+
+						func(em endpointmanager.EndpointManager) *ipcache.IPCache {
+							pr := policy.NewPolicyRepository(hivetest.Logger(t), nil, nil, nil, nil, api.NewPolicyMetricsNoop())
```

Comment:
Could just depend on `*slog.Logger` instead of creating a new one here, since you're passing in the logger at hive start time